model/assembly/CompiledTestConverterService.php

```diff
@@ -0,0 +1,108 @@
+use oat\oatbox\filesystem\File;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiTest\models\CompilationDataService;
+use oat\taoQtiTest\models\PhpCodeCompilationDataService;
+use oat\taoQtiTest\models\XmlCompilationDataService;
+
+class CompiledTestConverterService extends ConfigurableService
+{
+    const SERVICE_ID = 'taoDeliveryRdf/CompiledTestConverterService';
+
+    const OPTION_PHP_COMPILATION_SERVICE = 'php_compilation_service';
+    const OPTION_XML_COMPILATION_SERVICE = 'xml_compilation_service';
+
+    /**
+     * @var CompilationDataService
+     */
+    private $phpCompilationService = null;
+
+    /**
+     * @var CompilationDataService
+     */
+    private $xmlCompilationService = null;
+
+    /**
+     * CompiledTestConverterService constructor.
+     * @param array $options
+     */
+    public function __construct($options = array())
+    {
+        parent::__construct($options);
+
+        $this->phpCompilationService = $this->getOption(self::OPTION_PHP_COMPILATION_SERVICE);
+        if (!$this->phpCompilationService instanceof PhpCodeCompilationDataService) {
+            throw new InvalidArgumentException(sprintf('%s option must be an instance of %s', self::OPTION_PHP_COMPILATION_SERVICE,  PhpCodeCompilationDataService::class));
+        }
+
+        $this->xmlCompilationService = $this->getOption(self::OPTION_XML_COMPILATION_SERVICE);
+        if (!$this->xmlCompilationService instanceof XmlCompilationDataService) {
+            throw new InvalidArgumentException(sprintf('%s option must be an instance of %s', self::OPTION_XML_COMPILATION_SERVICE, XmlCompilationDataService::class));
+        }
+    }
+
+    /**
+     * @param File                                          $file
+     * @param tao_models_classes_service_StorageDirectory   $directory
+     * @return File
+     * @throws \common_Exception
+     */
+    public function convertPhpToXml(File $file, tao_models_classes_service_StorageDirectory $directory)
+    {
+        $fileName = pathinfo($file->getBasename(), PATHINFO_FILENAME);
+        $xmlFile = $directory->getFile($fileName . '.xml');
+        if ($xmlFile->exists()) {
+            $xmlFile->delete();
+        }
+
+        $object = $this->phpCompilationService->readCompilationData($directory, $fileName);
+        $this->xmlCompilationService->writeCompilationData($directory, $fileName, $object);
+
+        return $xmlFile;
+    }
+
+    /**
+     * @param File                                          $file
+     * @param tao_models_classes_service_StorageDirectory   $directory
+     * @return File
+     * @throws \common_Exception
+     */
+    public function convertXmlToPhp(File $file, tao_models_classes_service_StorageDirectory $directory)
+    {
+        $fileName = pathinfo($file->getBasename(), PATHINFO_FILENAME);
+
+        $phpPath = $fileName . '.php';
+        $phpFile = $directory->getFile($phpPath);
+
+        if ($phpFile->exists()) {

```

Comment:
Can you delete the file which you're using as a source only after the end of convert action?

Reply:
It's not a source. Method creates new file (result of conversion) and check's if it already exists. If it exists it deletes this file to be able to write new data into this file - result of conversion.